library/src/main/java/org/apache/apex/malhar/lib/state/managed/StateTracker.java

```diff
@@ -32,56 +32,53 @@
 import org.slf4j.LoggerFactory;
 
 import com.google.common.base.Preconditions;
+import com.google.common.collect.Sets;
 
 /**
  * Tracks the size of state in memory and evicts buckets.
  */
 class StateTracker extends TimerTask
 {
-  //bucket id -> bucket id & time wrapper
-  private final transient ConcurrentHashMap<Long, BucketIdTimeWrapper> bucketAccessTimes = new ConcurrentHashMap<>();
-
-  private transient ConcurrentSkipListSet<BucketIdTimeWrapper> bucketHeap;
-
   private final transient Timer memoryFreeService = new Timer();
 
   protected transient AbstractManagedStateImpl managedStateImpl;
 
+  private transient long lastUpdateAccessTime = 0;
+  private final transient Set<Long> accessedBucketIds = Sets.newHashSet();
+  private transient TreeSet<BucketIdTimeWrapper> bucketHeapForAccess = Sets.newTreeSet();
+  private transient TreeSet<BucketIdTimeWrapper> bucketHeapForRelease = Sets.newTreeSet();
+
```

Comment:
it is possible to it by a map from time to a list of bucket. but seems not simplify the code. The reason to use two lists in order two threads can run without concurrent control. the concurrent control only need to happen when merge/swap